editor/window_wrapper.cpp

```diff
@@ -0,0 +1,429 @@
+		Vector2 screen_ratio = Vector2(source_screen_rect.size) / Vector2(dest_screen_rect.size);
+
+		// The screen positioning may change, so remove the original screen position.
+		control_rect.position -= source_screen_rect.position;
+		control_rect = Rect2i(control_rect.position * screen_ratio, control_rect.size * screen_ratio);
+		control_rect.position += dest_screen_rect.position;
+
+		restore_window(control_rect, p_screen);
+	} else {
+		window->set_current_screen(p_screen);
+		set_window_enabled(true);
+	}
+}
+
+void WindowWrapper::set_window_title(const String p_title) {
+	if (!is_window_available()) {
+		return;
+	}
+	window->set_title(p_title);
+}
+
+void WindowWrapper::set_margins_enabled(bool p_enabled) {
+	if (!is_window_available()) {
+		return;
+	}
+
+	if (!p_enabled && margins) {
+		margins->queue_free();
+		margins = nullptr;
+	} else if (p_enabled && !margins) {
+		Size2 borders = Size2(4, 4) * EDSCALE;
+		margins = memnew(MarginContainer);
+		margins->add_theme_constant_override("margin_right", borders.width);
+		margins->add_theme_constant_override("margin_top", borders.height);
+		margins->add_theme_constant_override("margin_left", borders.width);
+		margins->add_theme_constant_override("margin_bottom", borders.height);
+
+		window->add_child(margins);
+		margins->set_anchors_and_offsets_preset(PRESET_FULL_RECT);
+	}
+}
+
+WindowWrapper::WindowWrapper() {
+	if (SceneTree::get_singleton()->get_root()->is_embedding_subwindows() || !EDITOR_GET("interface/multi_window/enable")) {
+		return;
+	}
+
+	window = memnew(Window);
+	window->set_wrap_controls(true);
+
+	add_child(window);
+	window->hide();
+
+	window->connect("close_requested", callable_mp(this, &WindowWrapper::set_window_enabled).bind(false));
+
+	ShortcutBin *capturer = memnew(ShortcutBin);
+	window->add_child(capturer);
+
+	window_background = memnew(Panel);
+	window_background->set_anchors_and_offsets_preset(PRESET_FULL_RECT);
+	window->add_child(window_background);
+}
+
+// ScreenSelect
+
+void ScreenSelect::_build_advanced_menu() {
+	// Clear old screen list.
+	while (screen_list->get_child_count(false) > 0) {
+		Node *child = screen_list->get_child(0);
+		screen_list->remove_child(child);
+		child->queue_free();
+	}
+
+	// Populate screen list.
+	const real_t height = real_t(get_theme_font_size("font_size")) * 1.5;
+
+	int current_screen = get_window()->get_current_screen();
+	for (int i = 0; i < DisplayServer::get_singleton()->get_screen_count(); i++) {
+		Button *button = memnew(Button);
+
+		Size2 screen_size = Size2(DisplayServer::get_singleton()->screen_get_size(i));
+		Size2 button_size = Size2(height * (screen_size.x / screen_size.y), height);
+		button->set_custom_minimum_size(button_size);
+		screen_list->add_child(button);
+
+		button->set_text(itos(i));
+		button->set_text_alignment(HORIZONTAL_ALIGNMENT_CENTER);
+		button->set_tooltip_text(vformat(TTR("Make this panel floating in the screen %d"), i));
```

Comment:
```suggestion
		button->set_tooltip_text(vformat(TTR("Make this panel floating in the screen %d."), i));
```
or, but not sure if it's correct:
```suggestion
		button->set_tooltip_text(vformat(TTR("Make this panel floating on screen %d."), i));
```